### ISSUE TITLE
ImportError: libgdal.so.31

### ISSUE THREAD

**Issue by taliboliver** [isce-framework/sweets#3]
Hi @scottstanie , 
I installed sweets within my already existing environment with isce3, compass and dolphin. During the installation I noticed that dolphin was re-installed to a different version (See log below).

```
Successfully built dolphin
Installing collected packages: dolphin
  Attempting uninstall: dolphin
    Found existing installation: dolphin 0.0.3.post1.dev13+g0e7bcd7
    Uninstalling dolphin-0.0.3.post1.dev13+g0e7bcd7:
      Successfully uninstalled dolphin-0.0.3.post1.dev13+g0e7bcd7
Successfully installed dolphin-0.0.post1.dev369+gb9902e4
```
Installation when through without any error, however when testing sweets i get the following:

```
 sweets --bbox -102.3407 31.9909 -101.9407 32.3909 --start "2022-10-15" --track 78
Traceback (most recent call last):
  File "/home/cabrera/python/miniconda3/envs/opera/bin/sweets", line 8, in <module>
    sys.exit(main())
  File "/home/cabrera/python/miniconda3/envs/opera/lib/python3.9/site-packages/sweets/cli.py", line 133, in main
    from sweets.core import Workflow
  File "/home/cabrera/python/miniconda3/envs/opera/lib/python3.9/site-packages/sweets/core.py", line 19, in <module>
    from ._burst_db import get_burst_db
  File "/home/cabrera/python/miniconda3/envs/opera/lib/python3.9/site-packages/sweets/_burst_db.py", line 11, in <module>   
    from .utils import get_cache_dir
  File "/home/cabrera/python/miniconda3/envs/opera/lib/python3.9/site-packages/sweets/utils.py", line 8, in <module>
    from shapely import from_geojson, from_wkt, geometry
ImportError: cannot import name 'from_geojson' from 'shapely' (/home/cabrera/python/miniconda3/envs/opera/lib/python3.9/site-packages/shapely/__init__.py)
```
Also, now after the installation isce3 cannot find libgdal.so.31:
```
s1_geocode_stack.py -h
Traceback (most recent call last):
  File "/mnt/aurora-r0/cabrera/software/tools/COMPASS/src/compass/s1_geocode_stack.py", line 12, in <module>
    from s1reader.s1_orbit import get_orbit_file_from_dir, parse_safe_filename
  File "/mnt/aurora-r0/cabrera/software/tools/s1-reader/src/s1reader/__init__.py", line 5, in <module>
    from s1reader.s1_burst_slc import Sentinel1BurstSlc
  File "/mnt/aurora-r0/cabrera/software/tools/s1-reader/src/s1reader/s1_burst_slc.py", line 8, in <module>
    import isce3
  File "/mnt/aurora-r0/cabrera/software/tools/isce3/install/packages/isce3/__init__.py", line 2, in <module>
    from .ext import extisce3
  File "/mnt/aurora-r0/cabrera/software/tools/isce3/install/packages/isce3/ext/__init__.py", line 3, in <module>
    from . import isce3 as extisce3
ImportError: libgdal.so.31: cannot open shared object file: No such file or directory
```
 What do you think could have been the problem? 
Thanks so much in advance!

**Comment by scottstanie** [comment]
1. I removed the `from_geojson` function that only started in shapely 2.0 in favor of one that's more compatible: https://github.com/opera-adt/sweets/commit/8dc035fd7d00d89906128077ad86a77a5382478f

2. We figured out offline that this libgdal error was related to the conda version of isce3 getting installed over the manually-installed one. We worked offline to detangle some of the conda environement issues, where different gdal/isce versions were not playing nicely with each other. 
3. I'll be adding more detailed `dev` requirements in the future to warn against having an isce3 built from source with this one (conda doesn't detect when it's compiled and added manually to `LD_LIBRARY_PATH/PYTHONPATH`).

Keeping this open until I solve (3)
